src/Http/Middlewares/DeprecatedRoute.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Jobilla\DeprecatedRoutes\Http\Middlewares;
+
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Event;
+use Illuminate\Support\Facades\Log;
+use Jobilla\DeprecatedRoutes\Events\RouteIsDeprecated;
+
+class DeprecatedRoute
+{
+    /**
+     * @var string
+     */
+    public const HEADER_NAME = 'X-Deprecated-At';
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  mixed  $deprecatedAt
+     * @return mixed
+     */
+    public function handle(Request $request, \Closure $next, $deprecatedAt): Response
+    {
+        /** @var \Illuminate\Http\Response $response */
+        $response = $next($request);
+
+        $this->logDeprecatedRequest($request);
+        $this->fireDeprecationEvent($request, $response);
+
+        $response->header(static::HEADER_NAME, $this->getTimestampString($deprecatedAt));
+
+        return $response;
+    }
+
+    protected function logDeprecatedRequest(Request $request)
+    {
+        $level = Config::get('deprecated_routes.log_level');
```

Comment:
Should this have a default value, incase the config isn't loaded by the dev

Reply:
Isn't the config file in the package always loaded? And the `$this->mergeConfigFrom()` call in the service provider makes sure that any missing configs in user-land file will be filled in with the ones in the package's config file.

Reply:
Ah, didn't see the `mergeConfigFrom` line